frontend/src/metabase/admin/databases/containers/DatabaseEditApp.tsx

```diff
@@ -83,146 +71,140 @@ const mapStateToProps = (state: State) => {
 };
 
 const mapDispatchToProps = {
-  reset,
+  dismissSyncSpinner,
   initializeDatabase,
-  saveDatabase,
+  reset,
+  selectEngine,
   updateDatabase,
-  dismissSyncSpinner,
   deleteDatabase,
-  selectEngine,
-  onChangeLocation: push,
+  replace,
 };
 
-type DatabaseEditErrorType = {
-  data: {
-    message: string;
-    errors: { [key: string]: string };
-  };
-  statusText: string;
-  message: string;
-};
+// loads the database into the redux state and polls until the
+// database has been successfully synced for the first time
+// ideally this should be ported to RTKQuery
+function useDatabaseInitializer(
```

Comment:
Forgive me code reviewer for I have sinned. I'm trying to avoid a refactor to RTKQuery as part of this PR. ☹️